sources/packages/backend/apps/queue-consumers/src/processors/assessment/cancel-application-assessment.processor.ts

```diff
@@ -0,0 +1,83 @@
+import { Process, Processor } from "@nestjs/bull";
+import { Job } from "bull";
+import { CancelAssessmentQueueInDTO } from "@sims/services/queue";
+import {
+  DisbursementScheduleService,
+  WorkflowClientService,
+} from "@sims/services";
+import { QueueNames } from "@sims/utilities";
+import { StudentAssessmentService } from "../../services";
+import { DataSource } from "typeorm";
+import { ZeebeGRPCError, ZeebeGRPCErrorTypes } from "@sims/services/zeebe";
+
+/**
+ * Process the workflow cancellation.
+ */
+@Processor(QueueNames.CancelApplicationAssessment)
+export class CancelApplicationAssessmentProcessor {
+  constructor(
+    private readonly dataSource: DataSource,
+    private readonly workflowClientService: WorkflowClientService,
+    private readonly studentAssessmentService: StudentAssessmentService,
+    private readonly disbursementScheduleService: DisbursementScheduleService,
+  ) {}
+
+  /**
+   * Cancel the workflow instance and rollback overawards, if any.
+   * @param job information to perform the process.
+   * @returns log message.
+   */
+  @Process()
+  async cancelAssessment(
+    job: Job<CancelAssessmentQueueInDTO>,
+  ): Promise<string> {
+    await job.log(`Start processing, attempt ${job.attemptsMade + 1}.`);
+    const assessment = await this.studentAssessmentService.getAssessmentById(
+      job.data.assessmentId,
+    );
+    // Try to cancel the workflow if an workflow id is present.
+    // TODO: once the assessment has a proper status flag to be marked as completed we can check if the workflow needed to be cancelled.
+    if (assessment.assessmentWorkflowId) {
+      await job.log(`Found workflow id ${assessment.assessmentWorkflowId}.`);
+      try {
+        await this.workflowClientService.deleteApplicationAssessment(
+          assessment.assessmentWorkflowId,
+        );
+        await job.log(`Workflow instance successfully cancelled.`);
+      } catch (error: unknown) {
+        if (
+          error instanceof ZeebeGRPCError &&
+          error.code !== ZeebeGRPCErrorTypes.NOT_FOUND
+        ) {
+          // An unexpected error happen and the process must be aborted.
+          throw error;
+        }
+        // NOT_FOUND error means that the call to Camunda was successful but the workflow instance was not found.
+        await job.log(
```

Comment:
so, are we only expecting `ZeebeGRPCErrorTypes.NOT_FOUND` error be coming to this point?

Reply:
This is the only one that matters here. If we got a `NOT_FOUND` we should not abort the method and just let the rollback overaward be also called.
Any other exception will interrupt the process and the job will retry later.